src/explore-education-statistics-frontend/src/modules/table-tool/components/FixedHeaderGroupedDataTable.tsx

```diff
@@ -1,256 +1,85 @@
-                        },
-                      )}
-                      scope="col"
-                      key={key}
-                    >
-                      {column}
-                    </th>
-                  );
-                }),
-              )}
-          </tr>
-        </thead>
-        {rowGroups.map((group, groupIndex) => {
-          const groupKey = `${group.label}-${groupIndex}`;
-
-          return (
-            !isStickyHeader && (
-              <tbody key={groupKey} className={styles.borderBottom}>
-                {group.rows.map((row, rowIndex) => {
-                  const rowKey = `${groupKey}_${row.label}_${rowIndex}`;
-
-                  return (
-                    <tr key={rowKey}>
-                      {rowIndex === 0 && (
-                        <th
-                          scope="rowgroup"
-                          rowSpan={group.rows.length || 1}
-                          className={classNames(styles.borderBottom)}
-                        >
-                          {group.label}
-                        </th>
-                      )}
-                      <th
-                        scope="row"
-                        className={classNames(
-                          'govuk-table__cell--numeric',
-                          styles.borderRight,
-                        )}
-                      >
-                        {row.label}
-                      </th>
-                      {!isStickyColumn &&
-                        row.columnGroups.flatMap((colGroup, colGroupIndex) =>
-                          colGroup.map((column, columnIndex) => {
-                            const cellKey = `${rowKey}_${colGroupIndex}_${column}_${columnIndex}`;
-
-                            return (
-                              <td
-                                className={classNames(
-                                  'govuk-table__cell--numeric',
-                                  {
-                                    [styles.borderLeft]:
-                                      columnIndex === 0 && colGroupIndex > 0,
-                                  },
-                                )}
-                                key={cellKey}
-                              >
-                                {column}
-                              </td>
-                            );
-                          }),
-                        )}
-                    </tr>
-                  );
-                })}
-              </tbody>
-            )
+const FixedHeaderGroupedDataTable = forwardRef<HTMLElement, Props>(
+  (props, ref) => {
+    const { caption } = props;
+
+    const mainTableRef = useRef<HTMLTableElement>(null);
+    const headerTableRef = useRef<HTMLTableElement>(null);
+    const columnTableRef = useRef<HTMLTableElement>(null);
+    const intersectionTableRef = useRef<HTMLTableElement>(null);
+
+    const setStickyElementSizes = throttle(() => {
```

Comment:
OK, I've gotta do some reading to make sure you're not making this stuff up :)  _Looks up lodash throttle..._

Reply:
Yep pretty much, resizing could be called many hundreds of times unnecessarily as the user resizes, so it's just to avoid unnecessary DOM mutations. I'll add a comment just to make this a little clearer